lib/base/scriptframe.hpp

```diff
@@ -29,7 +29,7 @@ struct ScriptFrame
 	static ScriptFrame *GetCurrentFrame();
 
 private:
-	static boost::thread_specific_ptr<std::stack<ScriptFrame *> > m_ScriptFrames;
+	static boost::thread_specific_ptr<std::vector<ScriptFrame*>> m_ScriptFrames;
```

Comment:
The underlying container of `std::stack` is [`std::deque`](https://en.cppreference.com/w/cpp/container/deque) by default, which is quite fast. When your motivation in changing this to `std::vector` is to gain more speed, then I wouldn't expect any improvements. It could make things even worse if the vector has to be reallocated so often.

Reply:
Why do you think "the vector has to be reallocated so often"? IIRC once a vector reaches its peak capacity, it's not reallocated until .shrink_to_fit().

Reply:
> it's not reallocated until .shrink_to_fit().

This method is used to shrink the vector, not to expand it! *Extra memory can be returned to the system via a call to [shrink_to_fit()](https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit)[[1]](https://en.cppreference.com/w/cpp/container/vector#cite_note-1).*

> Why do you think "the vector has to be reallocated so often"?

Do you know the exact size of that stack within a particular thread?

> Reallocations are usually costly operations in terms of performance. The [reserve()](https://en.cppreference.com/w/cpp/container/vector/reserve) function can be used to eliminate reallocations if the number of elements is known beforehand.

I guess not, so you won't be able to reserve the capacity in advance and if there are way too many frames stored on that stack, the vector will have to reallocate its internal memory each time and copy all existing items into the new one. I see no reason in changing this, as its default container is sufficient for its intended purpose.

Reply:
No, it doesn't reallocate **each** time, even if new items are inserted: https://stackoverflow.com/a/33571216

In case of deletion it's even not reallocated at all: https://stackoverflow.com/a/50599784